src/search.rs

```diff
@@ -1274,4 +1366,110 @@ mod tests {
 
         Ok(())
     }
+
+    /// enable vector searching and configure an userProvided embedder
+    async fn setup_hybrid_searching(client: &Client, index: &Index) -> Result<(), Error> {
+        use crate::settings::{Embedder, UserProvidedEmbedderSettings};
+        let features = crate::features::ExperimentalFeatures::new(&client)
+            .set_vector_store(true)
+            .update()
+            .await
+            .expect("could not enable the vector store");
+        assert_eq!(features.vector_store, true);
+        let embedder_setting = Embedder::UserProvided(UserProvidedEmbedderSettings { dimensions: 1 });
+        let t3 = index
+            .set_settings(&crate::settings::Settings {
+                embedders: Some(HashMap::from([("default".to_string(), embedder_setting)])),
+                ..crate::settings::Settings::default()
+            })
+            .await?;
+        t3.wait_for_completion(&client, None, None).await?;
+        Ok(())
+    }
+
+    #[meilisearch_test]
+    async fn test_with_vectors(client: Client, index: Index) -> Result<(), Error> {
+        setup_hybrid_searching(&client, &index).await?;
+        setup_test_index(&client, &index).await?;
+
+        let results: SearchResults<Document> = index.search()
+            .with_query("lorem ipsum")
+            .with_retrieve_vectors(true)
+            .execute().await?;
+        assert_eq!(results.hits.len(), 1);
+        let expected = Vectors::from(&[1000.0]);
+        assert_eq!(results.hits[0].result._vectors, Some(expected));
+
+        let results: SearchResults<Document> = index.search()
+            .with_query("lorem ipsum")
+            .with_retrieve_vectors(false)
+            .execute().await?;
+        assert_eq!(results.hits.len(), 1);
+        assert_eq!(results.hits[0].result._vectors, None);
+        Ok(())
+    }
+
+    #[meilisearch_test]
+    async fn test_hybrid(client: Client, index: Index) -> Result<(), Error> {
```

Comment:
Small status update:
- [x] `with_vectors`
- [x] Ollama
- [x] Rest
- [ ] refactoring `test_hybrid` to mock the network calls made instead of failing to figgure out how the local embedder really works